src/core/services/interaction/interaction.js

```diff
@@ -129,4 +132,18 @@ MdInteractionService.prototype.onBufferInputEvent = function(event) {
  */
 MdInteractionService.prototype.getLastInteractionType = function() {
   return this.lastInteractionType;
-};
\ No newline at end of file
+};
+
+/**
+ * @ngdoc method
+ * @name $mdInteraction#isProgrammatic
+ * @description Method to detect whether any interaction happened recently.
+ * @param {number=} checkDelay Time to check for any interaction to have been triggered.
+ * @returns {boolean} Whether there was any interaction or not.
+ */
+MdInteractionService.prototype.isProgrammatic = function(checkDelay) {
```

Comment:
I know that we decided on the naming for this together, but I'm still not sure that `isProgrammatic` is very obvious. It seems like it implies that the service is programmatic. That being said, I don't have any better suggestions at the moment, perhaps other team member can pitch in?